open-sphere-base/mantle/src/main/java/io/opensphere/mantle/icon/impl/gui/IconManagerFrame.java

```diff
@@ -256,48 +272,47 @@ private void addSelectedToFavorites()
      */
     private void createIconPopupMenuItems(JPopupMenu puMenu)
     {
-        JMenuItem addToFavoritesMI = new JMenuItem("Add Selected To Favorites");
-        addToFavoritesMI.addActionListener(e -> addSelectedToFavorites());
-        puMenu.add(addToFavoritesMI);
-
-        JMenuItem rotateSelectedMI = new JMenuItem("Rotate Selected...");
-        rotateSelectedMI.addActionListener(e -> rotateSelected());
-        puMenu.add(rotateSelectedMI);
+        ArrayList<JMenuItem> menuBarList = getSharedButtons();
+        menuBarList.forEach(button -> puMenu.add(button));
 
-        JMenuItem deleteSelectedMI = new JMenuItem("Delete Selected...");
+        JMenuItem deleteSelectedMI = new JMenuItem("Delete Selected");
         deleteSelectedMI.addActionListener(e -> deleteSelected());
         puMenu.add(deleteSelectedMI);
-
-        JMenuItem deSelectAllMI = new JMenuItem("De-Select All");
-        deSelectAllMI.addActionListener(e -> deselectAll());
-        puMenu.add(deSelectAllMI);
     }
 
     /**
      * Creates the menu bar menu items, adding the newly created items to the
      * {@link #myEditMenu}.
      */
     private void createMenuBarEditMenuItems()
+    {
+        ArrayList<JMenuItem> menuBarList = getSharedButtons();
+        menuBarList.forEach(button -> myEditMenu.add(button));
+
+        JMenuItem deleteSelectedTreeNodeMI = new JMenuItem("Delete Selected Icon Set");
+        deleteSelectedTreeNodeMI.addActionListener(e -> deleteSelectedTreeNode());
+        myEditMenu.add(deleteSelectedTreeNodeMI);
+    }
+
+    /**
+     * Creates the menu bar and popup buttons for
+     * {@link #createMenuBarFileMenuItems()} {@link #createIconPopupMenuItems(JPopupMenu)}.
+     * @return buttonList
+     */
+    private ArrayList<JMenuItem> getSharedButtons()
     {
         JMenuItem addToFavoritesMI = new JMenuItem("Add Selected To Favorites");
         addToFavoritesMI.addActionListener(e -> addSelectedToFavorites());
-        myEditMenu.add(addToFavoritesMI);
 
-        JMenuItem rotateSelectedMI = new JMenuItem("Rotate Selected...");
+        JMenuItem rotateSelectedMI = new JMenuItem("Rotate Selected");
         rotateSelectedMI.addActionListener(e -> rotateSelected());
-        myEditMenu.add(rotateSelectedMI);
-
-        JMenuItem deleteSelectedMI = new JMenuItem("Delete Selected...");
-        deleteSelectedMI.addActionListener(e -> deleteSelected());
-        myEditMenu.add(deleteSelectedMI);
-
-        JMenuItem deleteSelectedTreeNodeMI = new JMenuItem("Delete Selected Icon Set...");
-        deleteSelectedTreeNodeMI.addActionListener(e -> deleteSelectedTreeNode());
-        myEditMenu.add(deleteSelectedTreeNodeMI);
 
         JMenuItem deSelectAllMI = new JMenuItem("De-Select All");
         deSelectAllMI.addActionListener(e -> deselectAll());
-        myEditMenu.add(deSelectAllMI);
+
+        ArrayList<JMenuItem> buttonList = new ArrayList<JMenuItem>(Arrays.asList(addToFavoritesMI, rotateSelectedMI, deSelectAllMI));

```

Comment:
lets get rid of this line and turn it into a New.List() with values. If you look up the doc for New.List() there is a method for creating a new list with values

Reply:
so we can get rid of the buttonList variable and just do `return New.List(value1,value2,value3,value4,value1million)`

Reply:
changed